update dependencies and fix build

patrickkettner opened the pull request:
While working on #159, I figured you may want to update the dependencies and build stuff to move the broken/deprecated bits to the current versions. Testing as working for me locally.

patrickkettner committed:
update dependencies and fix build

kylefowler commented:
running into a few build errors on the main project related to `@types/file-type` not needing to be in the dependencies and then a few typescript compile issues

patrickkettner committed:
fix main project build

patrickkettner commented:
Thanks @kylefowler!
It actually looks like that is true in the master branch today (I didn't touch `file-type` or its types). 
Since that [types file is deprecated ](https://www.npmjs.com/package/@types/file-type), I removed it. After that, there were a handful of changes to the typescript needed. `html-webpack-inline-source-plugin` was deprecated and didn't support webpack 5, so I replaced it with that projects recommended successor.

kylefowler reviewed (approved):
Thanks for taking care of the main build
